### PR TITLE
Add openapi.yaml and update README

### PR DESCRIPTION
**Pull request by knikolla**
Provided a minimum of documentation to reduce knowledge debt and technical debt.

### CODE REVIEW

**Comment by larsks** [comment]
I feel like the pr description suffers from an order-of-operations problem and needs parentheses for clarity...

> Provided a minimum of documentation to reduce (knowledge and technical debt).

:smile:

File: README.md [hunk 1]
```diff
@@ -1,2 +1,23 @@
...
+The plugin can be enabled by adding `coldfront_plugin_api` to `ENABLED_APPS`
+in the Django `local_settings.py`.
+
+If the environment variable `PLUGIN_AUTH_OIDC` is detected, authentication
+will be done through `mozilla-django-oidc`, using the same configuration
+as the rest of ColdFront.
+
+**Note**: If using service accounts and Keycloak, it is necessary to add
+`openid` to the client scope of the service account performing the API
+request. This is because  the `mozilla-django-oidc` Django Rest Framework
```
Comment (larsks) [inline README.md:19]: Is this a change that needs to be made in Keycloak? We should clarify that here.

Comment (knikolla) [reply README.md:19]: It's a change that needs to be made through the Keycloak dashboard when configuring the service account. https://access.redhat.com/documentation/en-us/red_hat_single_sign-on_continuous_delivery/2/html/server_administration_guide/clients#client_scopes

Comment (knikolla) [reply README.md]: Updated to include link to the above documentation.

**Review by larsks — APPROVED** [review]

**Pull request merged**